core/domain/acl_decorators.py

```diff
@@ -323,20 +323,48 @@ def test_can_access(self, **kwargs):
     return test_can_access
 
 
-def can_comment_on_feedback_thread(handler):
-    """Decorator to check whether the user can view feedback for a given
-    exploration.
+def can_view_feedback_thread(handler):
+    """Decorator to check whether the user can view feedback thread.
     """
 
-    def test_can_access(self, exploration_id, **kwargs):
-        """Checks if the user can view the exploration feedback.
+    def test_can_access(self, thread_id, **kwargs):
+        """Checks if the user can view the feedback thread.
 
         Args:
-            exploration_id: str. The exploration id.
+            thread_id: str. The feedback thread id.
+        """
+        exploration_id = thread_id.split('.')[0]
+        if exploration_id in feconf.DISABLED_EXPLORATION_IDS:
+            raise base.UserFacingExceptions.PageNotFoundException
+
+        exploration_rights = rights_manager.get_exploration_rights(
+            exploration_id, strict=False)
+        if rights_manager.check_can_access_activity(
+                self.user, exploration_rights):
+            return handler(self, thread_id, **kwargs)
+        else:
+            raise self.UnauthorizedUserException(
+                'You do not have credentials to view exploration feedback.')
+    test_can_access.__wrapped__ = True
+
+    return test_can_access
+
+
+def can_send_message_to_thread(handler):
```

Comment:
Doesn't sound correct. What was the issue with can_comment_on_feedback_thread?

Reply:
edited